EssentialApp/EssentialAppTests/FeedUIIntegrationTests.swift

```diff
@@ -210,6 +210,17 @@ final class FeedUIIntegrationTests: XCTestCase {
 		XCTAssertEqual(view1?.isShowingRetryAction, false, "Expected no retry action for second view on retry")
 	}
 	
+	func test_feedImageView_forwardsFeedImageIDOnTapWhenImageHasLoadSuccessfully() {
+		let image = makeImage()
+		var capturedImages = [FeedImage]()
+		let (sut, loader) = makeSUT(imageHandler: { capturedImages.append($0) })
+		
+		sut.loadViewIfNeeded()
+		loader.completeFeedLoading(with: [image])
+		sut.simulateTapOnFeedImageView(at: 0)
+		XCTAssertEqual(capturedImages, [image], "Expected tapped image to be forwarded without changes")
```

Comment:
Missing pre-condition assertion (no image should have been selected). 

```suggestion
		sut.loadViewIfNeeded()
		loader.completeFeedLoading(with: [image])
		XCTAssertEqual(capturedImages, [])
		
		sut.simulateTapOnFeedImageView(at: 0)
		XCTAssertEqual(capturedImages, [image], "Expected tapped image to be forwarded without changes")
```

Also, you should add at least 2 images and select both at different indexes to increase coverage. One is not enough.